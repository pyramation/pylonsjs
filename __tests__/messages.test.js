import { coin } from '@cosmjs/amino';
import { Pylonstech } from '../src/proto';

const messages = Pylonstech.pylons.pylons.MessageComposer.withTypeUrl;

const { createAccount, createCookbook } = messages;

it('createAccount', async () => {
  expect(
    createAccount({
      creator: 'creator1',
      username: 'myusername'
    })
  ).toMatchSnapshot();
});

it('createCookbook', async () => {
  expect(
    createCookbook({
      ID: 'someId',
      creator: 'creator1',
      description: 'desc',
      developer: 'somedev',
      enabled: true,
      name: 'somename',
      supportEmail: 'yolo@gmail.com',
      version: 'v1'
    })
  ).toMatchSnapshot();
});
