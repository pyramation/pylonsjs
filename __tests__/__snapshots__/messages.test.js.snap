// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createAccount 1`] = `
Object {
  "typeUrl": "/Pylonstech.pylons.pylons.MsgCreateAccount",
  "value": Object {
    "creator": "creator1",
    "username": "myusername",
  },
}
`;

exports[`createCookbook 1`] = `
Object {
  "typeUrl": "/Pylonstech.pylons.pylons.MsgCreateCookbook",
  "value": Object {
    "ID": "someId",
    "creator": "creator1",
    "description": "desc",
    "developer": "somedev",
    "enabled": true,
    "name": "somename",
    "supportEmail": "yolo@gmail.com",
    "version": "v1",
  },
}
`;
